api/product.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	"fmt"
+	"github.com/labstack/echo/v4"
+	"HelloWorld/models"
+	"net/http"
+)
+
+type productManager interface {
+	Add(product *models.Product) error
+	ReadAll(c echo.Context) ([]models.Product, error)
+}
+
+type ProductRouter struct {
+	productManager productManager
+}
+
+func NewProductRouter(productMgr productManager) *ProductRouter {
+	return &ProductRouter{
+		productManager: productMgr,
+	}
+}
+
+// CreateProduct
+func (p *ProductRouter) CreateProduct(c echo.Context) error {
```

Comment:
Rename it from `CreateProduct` to just `Create` cuz you already inside product manager, no need to replicate the words